src/react/components/pages/projectSettings/projectForm.test.tsx

```diff
@@ -18,167 +19,204 @@ describe("Project Form Component", () => {
-            expect(wrapper.state().formData.sourceConnectionId).toEqual(project.sourceConnectionId);
-            expect(wrapper.find("select#root_sourceConnectionId").exists()).toBe(true);
-            wrapper.find("select#root_sourceConnectionId").simulate("change", { target: { value: newConnectionId } });
-
-            expect(wrapper.state().formData.sourceConnectionId).toEqual(newConnectionId);
-            const form = wrapper.find("form");
-            form.simulate("submit");
-            expect(onSubmit).toBeCalledWith({
-                ...project,
-                tags: JSON.stringify(project.tags),
                 sourceConnectionId: newConnectionId,
-            });
-
-        });
-
-        it("should update target connection ID upon submission", () => {
-            const wrapper = createComponent({
-                project,
-                connections,
-                onSubmit,
-            });
-            const newConnectionId = connections[1].id;
-            expect(wrapper.state().formData.targetConnectionId).not.toEqual(newConnectionId);
-            expect(wrapper.state().formData.targetConnectionId).toEqual(project.targetConnectionId);
-            expect(wrapper.find("select#root_targetConnectionId").exists()).toBe(true);
-            wrapper.find("select#root_targetConnectionId").simulate("change", { target: { value: newConnectionId } });
-            expect(wrapper.state().formData.targetConnectionId).toEqual(newConnectionId);
-            const form = wrapper.find("form");
-            form.simulate("submit");
-            expect(onSubmit).toBeCalledWith({
-                ...project,
-                tags: JSON.stringify(project.tags),
                 targetConnectionId: newConnectionId,
-            });
-        });
-
-        it("should call onChangeHandler on submission with stringified tags", () => {
-            const wrapper = createComponent({
-                project,
-                connections,
-                onSubmit,
-            });
-            const form = wrapper.find("form");
-            form.simulate("submit");
-            expect(onSubmit).toBeCalledWith({
-                ...project,
-                tags: JSON.stringify(project.tags),
-            });
+                description: newDescription,
+                tags: expect.arrayContaining([
+                    ...project.tags,
+                    {
+                        name: newTagName,
+                        color: expect.stringMatching(/^#([0-9a-fA-F]{3}){1,2}$/i),
+                    },
+                ]),
+            }),
+        );
+    });
+
+    it("empty project has initial state loaded correctly", () => {
+        const onSubmit = jest.fn();
+        const wrapper = createComponent({
+            project: null,
+            connections,
+            onSubmit,
         });
+        expect(wrapper.state().formData.name).toBe(undefined);
```

Comment:
Nit: May be worth setting a variable to `wrapper.state()` so you don't have to call it within each expectation.